### PR TITLE
Issue 1886: Introduces readRange method in RevisionStreamClient

### LINKED ISSUE #1886: RevisionedStreamClient should have a readRange() call
**Problem description**
Currently RevisionedStreamClient has `readFrom(Revision start)` which allows for reading all events after a revision, but there is no way to tell it to stop at a particular revision. 

**Problem location**
RevisionedStreamClient 

**Suggestions for an improvement**
We should add an option to have an API like `readRange(Revision start, Revision end)` that will stop when it gets to the revision provided for `end`.

### PR DESCRIPTION
**Pull request by a6dulaleem**
**Change log description**  
This PR introduces a new method that allows reading data between two specific revisions - a start revision and an end revision.

**Purpose of the change**  
Fixes #1886

**What the code does**  
Reads all data from  a given start revision to the end revision and also adds unit test for the newly added method.

**How to verify it**  
All test should pass

**Note :-**  This PR is related to this draft PR https://github.com/pravega/pravega/pull/6942. for more context please have a look to this.

### CODE REVIEW

File: client/src/main/java/io/pravega/client/state/impl/RevisionedStreamClientImpl.java [hunk 1]
```diff
@@ -149,6 +149,27 @@ public Iterator<Entry<Revision, T>> readFrom(Revision revision) {
             return new StreamIterator(startOffset, endOffset);
         }
     }
+
+    @Override
+    public Iterator<Entry<Revision, T>> readRange(Revision startRevision, Revision endRevision) {
+        log.trace("Read segment {} from revision {} to revision {}", segment, startRevision, endRevision);
+        synchronized (lock) {
+            long startOffset = startRevision.asImpl().getOffsetInSegment();
```
[the remark this answers is not in the document]
Comment (a6dulaleem) [reply client/src/main/java/io/pravega/client/state/impl/RevisionedStreamClientImpl.java:160]: Added :)

**Review by kotlasaicharanreddy — APPROVED** [review]

File: client/src/main/java/io/pravega/client/state/impl/RevisionedStreamClientImpl.java [hunk 2]
```diff
@@ -149,6 +149,27 @@ public Iterator<Entry<Revision, T>> readFrom(Revision revision) {
...
+            SegmentInfo segmentInfo = Futures.getThrowingException(meta.getSegmentInfo());
+            long endOffset = endRevision.asImpl().getOffsetInSegment();
+            if (startOffset < segmentInfo.getStartingOffset()) {
+                throw new TruncatedDataException(format("Data at the supplied revision {%s} has been truncated. The current segment info is {%s}", startRevision, segmentInfo));
+            }
+            if (startOffset == endOffset) {
+                log.debug("No new updates to be read from revision {}", endRevision);
```
[the remark this answers is not in the document]
Comment (a6dulaleem) [reply client/src/main/java/io/pravega/client/state/impl/RevisionedStreamClientImpl.java]: I am following the same pattern as readFrom() is using. so for the same test its using debug.

**Review by anju-c-das — APPROVED** [review]
LGTM

**Review by ShwethaSNayak — APPROVED** [review]
LGTM

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**